install/kubernetes/cilium/charts/config/templates/configmap.yaml

```diff
@@ -484,9 +484,13 @@ data:
   k8s-require-ipv4-pod-cidr: {{ .Values.global.ipv4.enabled | quote}}
 {{- end }}
 {{- end }}
+{{- if and .Values.global.k8s .Values.global.k8s.requireIPv6PodCIDR }}
+  k8s-require-ipv6-pod-cidr: {{ .Values.global.k8s.requireIPv6PodCIDR | quote}}
+{{- else }}
 {{- if eq $ipam "cluster-pool" }}
   k8s-require-ipv6-pod-cidr: {{ .Values.global.ipv6.enabled | quote}}
 {{- end }}
+{{- end }}
```

Comment:
Any way this construction will never work if you set `--set global.k8s.requireIPv6PodCIDR=false`, then `if .Values.global.k8s.requireIPv6PodCIDR` will also return false